src/cursor/cur_stat.c

```diff
@@ -606,24 +606,37 @@ __wt_curstat_open(WT_SESSION_IMPL *session,
 		    session, &cval, "all", &sval)) == 0 && sval.val != 0) {
 			if (!FLD_ISSET(conn->stat_flags, WT_CONN_STAT_ALL))
 				goto config_err;
-			F_SET(cst, WT_CONN_STAT_ALL | WT_CONN_STAT_FAST);
+			F_SET(cst, WT_CONN_STAT_ALL |
+			    WT_CONN_STAT_CACHE_WALK | WT_CONN_STAT_FAST);
 		}
 		WT_ERR_NOTFOUND_OK(ret);
 		if ((ret = __wt_config_subgets(
 		    session, &cval, "fast", &sval)) == 0 && sval.val != 0) {
 			if (F_ISSET(cst, WT_CONN_STAT_ALL))
 				WT_ERR_MSG(session, EINVAL,
-				    "only one statistics configuration value "
-				    "may be specified");
+				    "Only one of all, fast, none "
+				    "configuration values should be specified");
 			F_SET(cst, WT_CONN_STAT_FAST);
 		}
 		WT_ERR_NOTFOUND_OK(ret);
+
+		if ((ret = __wt_config_subgets(session,
+		    &cval, "cache_walk", &sval)) == 0 && sval.val != 0) {
+			if (!F_ISSET(cst, WT_CONN_STAT_ALL | WT_CONN_STAT_FAST))
+				WT_ERR_MSG(session, EINVAL,
+				    "the value \"cache_walk\" is not "
+				    "compatible with \"none\" statistics "
```

Comment:
If I provide "cache_walk" only, and don't necessarily provide "none", it will still get set to none and error out saying "not compatible with none statistics configuration". Do you instead want to say on the lines of "cache_walk can only be provided with either fast or all".